src/segment_brewfather.go

```diff
@@ -0,0 +1,341 @@
+	BFStatusFermenting   string = "Fermenting"
+	BFStatusConditioning string = "Conditioning"
+	BFStatusCompleted    string = "Completed"
+	BFStatusArchived     string = "Archived"
+)
+
+// Returned from https://api.brewfather.app/v1/batches/batch_id/readings
+type BatchReading struct {
+	Comment     string  `json:"comment"`
+	Gravity     float64 `json:"sg"`
+	DeviceType  string  `json:"type"`
+	DeviceID    string  `json:"id"`
+	Temperature float64 `json:"temp"`      // celsius - need to add F conversion
+	Timepoint   int64   `json:"timepoint"` // << check what these are...
+	Time        int64   `json:"time"`      // <<
+}
+type Batch struct {
+	// Json tagged values returned from https://api.brewfather.app/v1/batches/batch_id
+	Status string `json:"status"`
+	Recipe struct {
+		Name string `json:"name"`
+	} `json:"recipe"`
+	BrewDate         int64 `json:"brewDate"`
+	FermentStartDate int64 `json:"fermentationStartDate"`
+	BottlingDate     int64 `json:"bottlingDate"`
+
+	MeasuredOg  float64 `json:"measuredOg"`
+	MeasuredFg  float64 `json:"measuredFg"`
+	MeasuredAbv float64 `json:"measuredAbv"`
+
+	// copy of the latest BatchReading in here.
+	Reading *BatchReading
+
+	// Calculated values we need to cache because they require the rest query to reproduce
+	TemperatureTrend float64 // diff between this and last, short term trend
+}
+
+func (bf *brewfather) enabled() bool {
+	data, err := bf.getResult()
+	if err != nil {
+		return false
+	}
+	bf.Batch = *data
+
+	if bf.Batch.Reading != nil {
+		readingDate := time.UnixMilli(bf.Batch.Reading.Time)
+		bf.ReadingAge = int(time.Since(readingDate).Hours())
+	} else {
+		bf.ReadingAge = -1
+	}
+
+	bf.TemperatureTrendIcon = bf.getTrendIcon(bf.TemperatureTrend)
+	bf.StatusIcon = bf.getBatchStatusIcon(data.Status)
+
+	fermStartDate := time.UnixMilli(bf.Batch.FermentStartDate)
+	bottlingDate := time.UnixMilli(bf.Batch.BottlingDate)
+
+	switch bf.Batch.Status {
+	case BFStatusFermenting:
+		// in the fermenter now, so relative to today.
+		bf.DaysFermenting = uint(time.Since(fermStartDate).Hours() / 24)
+		bf.DaysBottled = 0
+		bf.DaysBottledOrFermented = &bf.DaysFermenting
+	case BFStatusConditioning, BFStatusCompleted, BFStatusArchived:
+		bf.DaysFermenting = uint(bottlingDate.Sub(fermStartDate).Hours() / 24)
+		bf.DaysBottled = uint(time.Since(bottlingDate).Hours() / 24)
+		bf.DaysBottledOrFermented = &bf.DaysBottled
+	default:
+		bf.DaysFermenting = 0
+		bf.DaysBottled = 0
+		bf.DaysBottledOrFermented = nil
+	}
+
+	// URL property set to weblink to the full batch page
+	batchID := bf.props.getString(BFBatchID, "")
+	if len(batchID) > 0 {
+		bf.URL = fmt.Sprintf("https://web.brewfather.app/tabs/batches/batch/%s", batchID)
+	}
+
+	bf.DefaultString = bf.formatDefaultString()
+
+	return true
+}
+
+// build a default string for the segment using Sprintf for speed and to provide a simple starting point without complex template logic
+func (bf *brewfather) formatDefaultString() string {
```

Comment:
You could also add default template? I want to move away from the string based implementations. I see you have both but it doesn't seem like `DefaultString` is used anywhere, right? If it's up to me, I would not have this function and only rely on the default template.

Reply:
I pondered this one for a bit - figured a quick coded version that might be the golden path for many would be more efficient that trying to replicate with template logic.  Mainly coming at it from a perf angle, but I haven't profiled it.

It does feel a bit janky to have both.

I'll switch to default template for consistency.

Reply:
fixed and pushed